http/test/test_parser.ml

```diff
@@ -162,6 +162,26 @@ let parse_result_notifies_start_of_body () =
   [%test_result: string] ~expect:"foobar"
     (String.sub buf ~pos:count ~len:(String.length buf - count))
 
+let parse_proxy_get () =
+  let buf =
+    "GET http://example.com/foo.html HTTP/1.1\r\n\
+     Host: example.com\r\n\
+     Proxy-Authorization: Basic dXNlcjpwYXNz\r\n\
+     \r\n\
+     foobar"
+  in
+  let expected_req =
+    make_req
+      ~headers:
+        (Http.Header.of_list
+           [
+             ("Host", "example.com");
+             ("Proxy-Authorization", "Basic dXNlcjpwYXNz");
+           ])
+      `GET "http://example.com/foo.html"
+  in
+  assert_req_success ~here:[ [%here] ] ~expected_req ~expected_consumed:104 buf
```

Comment:
Could this test come in a separate PR? AFAIK, nothing in this PR was added to make this pass.

Reply:
Extracted to #1089.